clang/lib/StaticAnalyzer/Core/ExprEngineCXX.cpp

```diff
@@ -1233,3 +1233,34 @@ void ExprEngine::VisitAttributedStmt(const AttributedStmt *A,
 
   getCheckerManager().runCheckersForPostStmt(Dst, EvalSet, A, *this);
 }
+
+void ExprEngine::VisitCXXParenListInitExpr(const CXXParenListInitExpr *E,
+                                           ExplodedNode *Pred,
+                                           ExplodedNodeSet &Dst) {
```

Comment:
I've noticed that the `ExprEngine::VisitInitListExpr` does something really similar.
Would it make sense to hoist out the common parts and share the underlying implementation?

Reply:
Although feasible, the inconsistent handling of argument lists and compound types adds unnecessary complexity. It’s still worth considering as a future improvement. Probably would be better to keep it out of this issue to avoid extra changes. @steakhal @necto  WDYT?

Reply:
@necto I've updated tests to cover https://github.com/llvm/llvm-project/pull/148988#discussion_r2209440487